packages/block-library/src/social-links/edit.js

```diff
@@ -46,6 +49,22 @@ export function SocialLinksEdit( props ) {
 		setIconColor,
 	} = props;
 
+	const hasAnySelected = useSelect(
+		( select ) => {
+			const { getSelectedBlockClientId, getBlockOrder } =
+				select( blockEditorStore );
+			const selectedBlockClientId = getSelectedBlockClientId();
+			const innerBlockClientIds = getBlockOrder( clientId );
+
+			// Check if the selected block is the main block or any of its inner blocks
+			return (
+				selectedBlockClientId === clientId ||
+				innerBlockClientIds.includes( selectedBlockClientId )
+			);
+		},
+		[ clientId ]
+	);
```

Comment:
Why do you not use the `hasSelectedInnerBlock` + `isSelected` prop combo?

Reply:
Good question 😅 

Honestly I had a vague memory there was something "build-in" that handled this but I couldn't remember what it was and just wanted to land the experiment.

I'll refine this now so thank you for jogging my memory 🙇

Reply:
✅